src/devices/KeyMatrix/samples/Program.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Device.Gpio;
+using System.Linq;
+using System.Threading;
+using Iot.Device.KeyMatrix;
+
+Console.WriteLine("Please enter your output pins separated by a coma. For example: 27,22,25,6");
+var line = Console.ReadLine();
+IEnumerable<int> outputs = string.IsNullOrEmpty(line) ? new int[] { 27, 22, 25, 6 } : line.Split(',').Select(m => int.Parse(m));
+Console.WriteLine("Please enter your input pins separated by a coma. For example: 17,23,24,5");
+line = Console.ReadLine();
+IEnumerable<int> inputs = string.IsNullOrEmpty(line) ? new int[] { 17, 23, 24, 5 } : line.Split(',').Select(m => int.Parse(m));
+Console.WriteLine("Please enter the scanning interval in milliseconds. For example: 15");
+line = Console.ReadLine();
+int interval = int.TryParse(line, out int i) ? i : 15;
+Console.WriteLine("Please enter the number of keys you want to read individually events. For example: 20");
+line = Console.ReadLine();
+int count = int.TryParse(line, out int c) ? c : 20;
+
+// initialize keyboard
+KeyMatrix mk = new KeyMatrix(outputs, inputs, TimeSpan.FromMilliseconds(interval));
+
+// define the cancellation token.
+CancellationTokenSource source = new CancellationTokenSource();
+CancellationToken token = source.Token;
+
+// read key events
+for (int n = 0; n < count; n++)
+{
+    Console.WriteLine($"Waiting for matrix keyboard event... {n}/{count}");
+    KeyMatrixEvent? key = mk.ReadKey();
+    if (key is not object)
+    {
+        Console.WriteLine("No key pressed");
+        continue;
+    }
+
+    ShowKeyMatrixEvent(mk, key);
+}
+
+Console.WriteLine("This will now start listening to events and display them. Press a key to finish.");
+mk.KeyEvent += KeyMatrixEventReceived;
+mk.StartListeningKeyEvent();
+while (!Console.KeyAvailable)
```

Comment:
I'd prefer the key matrix APIs looked similar to console, so i.e.:

```csharp
while (keyMatrix.KeyAvailable)
{
  char key = keyMatrix.ReadKey();
}
```

treat as optional but I think having consistent APIs would be ideal

Reply:
it's not that easy to do. ReadKey() return a `KeyMatrixEvent` object which contains the coordinate of the key and the timeline. There are few differences with a keyboard which makes it complicated to implement:

- There is no key mapping available
- There is not key buffer, so you can't really detect if a key is pressed or not and then read the various keys

That would make the matrix class a bit more complex. Why not in the future if there is traction.

Reply:
sounds good